drivers/base/nmidebug/nmidebug.c

```diff
@@ -13,49 +12,31 @@
 
 /* FUNCTIONS ******************************************************************/
 
-PCHAR NmiBegin = "NMI4NMI@";
-
-FORCEINLINE
-VOID
-NmiClearFlag(VOID)
```

Comment:
What about that old code? Why didn't it work?

Reply:
The old code has a pointer dereference bug in two places that I found while running the code in debugger
https://github.com/reactos/reactos/blob/master/drivers/base/nmidebug/nmidebug.c#L28
https://github.com/reactos/reactos/blob/master/drivers/base/nmidebug/nmidebug.c#L46
That NMI callback is doing strange stuff that I didn't quite understand, it looks like just an incomplete code/driver.

Reply:
In any case, shouldn't one need to read/write to some IO port or do something for the CPU to ignore the NMI that has happened? (It seemed that the idea of the original code was to do that.) With the new code it appears that we break into the debugger (OK) but that's it.

Reply:
OK this commit gives slightly more context: https://github.com/reactos/reactos/commit/475b0fd7cb55b963b206d9eddcbfbb2e3bd88748

Reply:
The kernel trap handler is responsible for clearing the NMI iirc

Reply:
This code has a very specific purpose, which might not be obvious to the uninitiated reader, and that is the purpose :)